map/src/context/RoutingManager.js

```diff
@@ -102,60 +103,439 @@ function filterMode(data) {
+            profile = profile ?? routeProviders.profile;
+
+            const providers = this.allProviders();
+
+            if (!providers.find(r => r.key === router)) {
+                router = providers[0].key ?? 'osmand'; // fallback
+            }
+
+            const profiles = this.allProfiles(router);
+
+            if (!profiles.find(p => p.key === profile)) {
+                profile = profiles[0].key ?? 'car'; // fallback
+            }
+
+            if (router && profile) {
+                type = this.getProvider(router)?.type ?? 'osmand'; // fallback
+                // console.log('choosen:', type, router, profile);
+                return mergeStateObject(routeProviders, ctx.setRouteProviders, { type, router, profile });
+            }
+
+            return routeProviders; // unmodified
+        },
+
+        /*
+            providersOSRM and providersOsmAnd are loaded by loadRouterProviders()
+            Nevertheless, providersOsmAnd might be used before they are loaded
+            Therefore, OsmAnd must be initialized with minimal-fallback data
+        */
+        providersOSRM: [],
+        providersOsmAnd: [Object.assign({}, preloaded,
+            { key: preloaded.router, profiles: [{ key: 'car', name: 'Car' }] })],
+
     };
 }
 
 // load and validate OSRM and OsmAnd routing providers
 async function loadRouteProviders({ routeProviders, setRouteProviders, creatingRouteMode = null, setCreatingRouteMode = null }) {
+    const nameAliases = {
+        'ZLZK': 'OSRM',
+        'Routing OSM DE (Demo)': 'OSRM (backup)'
+    };
+
     // load OSRM providers first
     const osrm = await apiGet(`${process.env.REACT_APP_USER_API_SITE}/online-routing-providers.json`);
     if (osrm.ok) {
         try {
             const json = await osrm.json();
             if (json && json?.providers && json?.providers[0]?.name) {
+
+                json.providers.forEach(p => { // routes to profiles
+                    p.key = p.name; // name acts as key
+                    if (nameAliases[p.key]) {
+                        p.name = nameAliases[p.key];
+                    }
+                    if (p.routes) {
+                        p.routes.forEach(r => { // type to key
+                            r.key = r.type;
+                            delete r.type;
+                        });
+                        p.profiles = p.routes;
+                        delete p.routes;
+                    }
+                });
+
                 routeProviders = mergeStateObject(routeProviders, setRouteProviders, {
                     providersOSRM: json.providers,
-                    // name: 'OsmAnd Advanced Router', // debug
-                    name: json.providers[0].name, // set first OSRM provider
-                    profile: json.providers[0]?.routes[0]?.name // select first profile
+                    type: json.providers[0].type,
+                    router: json.providers[0].key, // select first OSRM key and type
```

Comment:
delete comments

Reply:
Refused: routeProviders is under development and we need all notes and reminders